src/ADAL.PCL.CoreCLR/CryptographyHelper.cs

```diff
@@ -25,14 +25,48 @@
 //
 //------------------------------------------------------------------------------
 
+using System;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+
 namespace Microsoft.IdentityModel.Clients.ActiveDirectory
 {
     internal class CryptographyHelper : ICryptographyHelper
     {
-
         public string CreateSha256Hash(string input)
         {
-            return null;
+            if (string.IsNullOrWhiteSpace(input))
+            {
+                return null;
+            }
+
+            using (var sha256 = SHA256.Create())
+            {
+                var inputBytes = Encoding.UTF8.GetBytes(input);
+                var outputBytes = sha256.ComputeHash(inputBytes);
+                return Convert.ToBase64String(outputBytes);
+            }
+        }
+
+        public byte[] SignWithCertificate(string message, X509Certificate2 certificate)
+        {
+            if (message == null)
+            {
+                throw new ArgumentNullException(nameof(message));
+            }
+
+            if (certificate == null)
+            {
+                throw new ArgumentNullException(nameof(certificate));
+            }
+
+            var privateKey = certificate.GetRSAPrivateKey();
+            var messageBytes = Encoding.UTF8.GetBytes(message);
+
+            // Using PKCS#1 padding scheme – same as in MSAL:
+            // https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/7fe94109/src/Microsoft.Identity.Client/Platforms/netstandard1.3/CryptographyHelper.cs#L68
+            return privateKey.SignData(messageBytes, HashAlgorithmName.SHA256, RSASignaturePadding.Pkcs1);
```

Comment:
This version doesn't dispose of the key, whereas the original version does.

As a rule of thumb, if you're copying existing code then it's generally a good idea to make as few changes as possible, particularly if they are stylistic or layout changes. It makes maintenance simpler in the future (easier when diff-ing the files to see what the material changes are).